File: POLIMORFISMO/EJ_3/EJ_3.py
class Cocinero:
    def __init__(self, nombre, sueldo_mes, horas_extra, sueldo_hora):
        self.nombre = nombre
        self.sueldo_mes = sueldo_mes
        self.horas_extra = horas_extra
        self.sueldo_hora = sueldo_hora

    def sueldo_total(self):
        return self.sueldo_mes + self.horas_extra * self.sueldo_hora

    def mostrar_si_sueldo_es(self, x):
        if self.sueldo_mes == x:
            print(f"Cocinero: {self.nombre}, SueldoMes: {self.sueldo_mes}")


class Mesero:
    def __init__(self, nombre, sueldo_mes, horas_extra, sueldo_hora, propina):
        self.nombre = nombre
        self.sueldo_mes = sueldo_mes
        self.horas_extra = horas_extra
        self.sueldo_hora = sueldo_hora
        self.propina = propina

    def sueldo_total(self):
        return self.sueldo_mes + self.horas_extra * self.sueldo_hora + self.propina

    def mostrar_si_sueldo_es(self, x):
        if self.sueldo_mes == x:
            print(f"Mesero: {self.nombre}, SueldoMes: {self.sueldo_mes}")


class Administrativo:
    def __init__(self, nombre, sueldo_mes, cargo):
        self.nombre = nombre
        self.sueldo_mes = sueldo_mes
        self.cargo = cargo

    def sueldo_total(self):
        return self.sueldo_mes

    def mostrar_si_sueldo_es(self, x):
        if self.sueldo_mes == x:
            print(f"Administrativo: {self.nombre}, SueldoMes: {self.sueldo_mes}")


c = Cocinero("Mario", 3000, 10, 20)
m1 = Mesero("Luis", 2500, 5, 15, 200)
m2 = Mesero("Juan", 2500, 8, 12, 150)
a1 = Administrativo("Ana", 3200.0, "Contadora")
a2 = Administrativo("Pepe", 2800.0, "Secretario")

print("Sueldos Totales:")
print(f"{c.nombre}: {c.sueldo_total()}")
print(f"{m1.nombre}: {m1.sueldo_total()}")
print(f"{m2.nombre}: {m2.sueldo_total()}")
print(f"{a1.nombre}: {a1.sueldo_total()}")
print(f"{a2.nombre}: {a2.sueldo_total()}")

print("\nEmpleados con sueldoMes == 2500:")
c.mostrar_si_sueldo_es(2500)
m1.mostrar_si_sueldo_es(2500)
m2.mostrar_si_sueldo_es(2500)
a1.mostrar_si_sueldo_es(2500)
a2.mostrar_si_sueldo_es(2500)

File: POLIMORFISMO/EJ_3/EJ_3.java
class Cocinero {
    String nombre;
    int sueldoMes;
    int horasExtra;
    float sueldoHora;

    public Cocinero(String nombre, int sueldoMes, int horasExtra, float sueldoHora) {
        this.nombre = nombre;
        this.sueldoMes = sueldoMes;
        this.horasExtra = horasExtra;
        this.sueldoHora = sueldoHora;
    }

    public float sueldoTotal() {
        return sueldoMes + (horasExtra * sueldoHora);
    }

    public void mostrarSiSueldoEs(int x) {
        if (sueldoMes == x) {
            System.out.println("Cocinero: " + nombre + ", SueldoMes: " + sueldoMes);
        }
    }
}

class Mesero {
    String nombre;
    int sueldoMes;
    int horasExtra;
    float sueldoHora;
    float propina;

    public Mesero(String nombre, int sueldoMes, int horasExtra, float sueldoHora, float propina) {
        this.nombre = nombre;
        this.sueldoMes = sueldoMes;
        this.horasExtra = horasExtra;
        this.sueldoHora = sueldoHora;
        this.propina = propina;
    }

    public float sueldoTotal() {
        return sueldoMes + (horasExtra * sueldoHora) + propina;
    }

    public void mostrarSiSueldoEs(int x) {
        if (sueldoMes == x) {
            System.out.println("Mesero: " + nombre + ", SueldoMes: " + sueldoMes);
        }
    }
}

class Administrativo {
    String nombre;
    float sueldoMes;
    String cargo;

    public Administrativo(String nombre, float sueldoMes, String cargo) {
        this.nombre = nombre;
        this.sueldoMes = sueldoMes;
        this.cargo = cargo;
    }

    public float sueldoTotal() {
        return sueldoMes;
    }

    public void mostrarSiSueldoEs(float x) {
        if (sueldoMes == x) {
            System.out.println("Administrativo: " + nombre + ", SueldoMes: " + sueldoMes);
        }
    }
}

public class Restaurante {
    public static void main(String[] args) {
        Cocinero c = new Cocinero("Mario", 3000, 10, 20);
        Mesero m1 = new Mesero("Luis", 2500, 5, 15, 200);
        Mesero m2 = new Mesero("Juan", 2500, 8, 12, 150);
        Administrativo a1 = new Administrativo("Ana", 3200f, "Contadora");
        Administrativo a2 = new Administrativo("Pepe", 2800f, "Secretario");

        System.out.println("Sueldos Totales:");
        System.out.println(c.nombre + ": " + c.sueldoTotal());
        System.out.println(m1.nombre + ": " + m1.sueldoTotal());
        System.out.println(m2.nombre + ": " + m2.sueldoTotal());
        System.out.println(a1.nombre + ": " + a1.sueldoTotal());
        System.out.println(a2.nombre + ": " + a2.sueldoTotal());

        System.out.println("\nEmpleados con sueldoMes == 2500:");
        c.mostrarSiSueldoEs(2500);
        m1.mostrarSiSueldoEs(2500);
        m2.mostrarSiSueldoEs(2500);
        a1.mostrarSiSueldoEs(2500f);
        a2.mostrarSiSueldoEs(2500f);
    }
}
